obspy/io/nied/knet/core.py

```diff
@@ -0,0 +1,312 @@
+    :type convert_stnm: bool
+    """
+    hdrdict = {'knet': {}}
+    hdrnames = ['Origin Time', 'Lat.', 'Long.', 'Depth. (km)', 'Mag.',
+                'Station Code', 'Station Lat.', 'Station Long.',
+                'Station Height(m)', 'Record Time', 'Sampling Freq(Hz)',
+                'Duration Time(s)', 'Dir.', 'Scale Factor', 'Max. Acc. (gal)',
+                'Last Correction', 'Memo.']
+    _i = 0
+    # Event information
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dt = flds[2] + ' ' + flds[3]
+    dt = UTCDateTime(datetime.strptime(dt, '%Y/%m/%d %H:%M:%S'))
+    # All times are in Japanese standard time which is 9 hours ahead of GMT
+    dt -= 9 * 3600.
+    hdrdict['knet']['evot'] = dt
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    lat = float(flds[1])
+    hdrdict['knet']['evla'] = lat
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    lon = float(flds[1])
+    hdrdict['knet']['evlo'] = lon
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dp = float(flds[2])
+    hdrdict['knet']['evdp'] = dp
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    mag = float(flds[1])
+    hdrdict['knet']['mag'] = mag
+
+    # Station information
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    # K-NET and KiK-Net station names can be more than 5 characters long
+    # which will cause the station name to be truncated when writing the
+    # the trace as miniSEED; if convert_stnm is enabled, the last two
+    # letters of the station code are written to the 'location' field
+    stnm = flds[2]
+    location = ''
+    if convert_stnm and len(stnm) > 5:
+        location = stnm[-2:]
+        stnm = stnm[:-2]
+    if len(stnm) > 7:
+        raise KnetFormatError("Station name can't be more than 7 characters\
+        long!")
+    hdrdict['station'] = stnm
+    hdrdict['location'] = location
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stla'] = float(flds[2])
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stlo'] = float(flds[2])
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stel'] = float(flds[2])
+
+    # Data information
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dt = flds[2] + ' ' + flds[3]
+    # A 15 s delay is added to the record time by the
+    # the K-NET and KiK-Net data logger
+    dt = UTCDateTime(datetime.strptime(dt, '%Y/%m/%d %H:%M:%S')) - 15.0
+    # All times are in Japanese standard time which is 9 hours ahead of GMT
+    dt -= 9 * 3600.
+    hdrdict['starttime'] = dt
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    freqstr = flds[2]
+    m = re.search('[0-9]*', freqstr)
+    freq = int(m.group())
+    delta = 1.0 / freq
+    hdrdict['delta'] = delta
+    hdrdict['sampling_rate'] = freq
```

Comment:
I don't think it's necessary to set both of these. They should be automatically related.